keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1462,225 +1278,309 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
-    // reserved for that Pod will not be started.
+// Constraint must have one and only one field set.
+type Constraint struct {
+    // All devices must have this attribute and its value must be the same.
+    //
+    // For example, if you specified "numa.dra.example.com" (a hypothetical example!),
+    // then only devices in the same NUMA node will be chosen.
     //
-    // There can be at most 32 such reservations. This may get increased in
-    // the future, but not reduced.
     // +optional
-    ReservedFor []ResourceClaimConsumerReference
+    // +listType=atomic
+    MatchAttribute *string
+
+    // Future extension, not part of the current design:
+    // A CEL expression which compares different devices and returns
+    // true if they match.
+    //
+    // Because it would be part of a one-of, old schedulers will not
+    // accidentally ignore this additional, for them unknown match
+    // criteria.
+    //
+    // matcher string
 }
 
-// ReservedForMaxSize is the maximum number of entries in
-// claim.status.reservedFor.
-const ResourceClaimReservedForMaxSize = 32
-```
+// Request is a request for one of many resources required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for one of several different alternatives.
+type Request struct {
+    // The name can be used to reference this request in a pod.spec.containers[].resources.claims
+    // entry.
+    //
+    // Must be a DNS label.
+    Name string
 
-##### ResourceClaimParameters
+    *ResourceRequestDetail
 
-```go
-// ResourceClaimParameters defines resource requests for a ResourceClaim in an
-// in-tree format understood by Kubernetes.
-type ResourceClaimParameters struct {
-    metav1.TypeMeta
-    // Standard object metadata
-    metav1.ObjectMeta
+    // FUTURE EXTENSION:
+    //
+    // OneOf contains a list of requests, only one of which must be satisfied.
+    // Requests are listed in order of priority.
+    //
+    // +optional
+    // +listType=atomic
+    // OneOf []ResourceRequestDetail
+}
 
-    // If this object was created from some other resource, then this links
-    // back to that resource. This field is used to find the in-tree representation
-    // of the claim parameters when the parameter reference of the claim refers
-    // to some unknown type.
-    GeneratedFrom *ResourceClaimParametersReference
+type ResourceRequestDetail struct {
+    // When referencing a DeviceClass, a request inherits additional
+    // configuration and requirements.
+    //
+    // +optional
+    DeviceClassName *string
 
-    // Shareable indicates whether the allocated claim is meant to be shareable
-    // by multiple consumers at the same time.
-    Shareable bool
+    // Config defines configuration parameters that apply to the requested resource(s).
+    // They are ignored while allocating the claim.
+    //
+    // +optional
+    // +listType=atomic
+    Config []ConfigurationParameters
```

Comment:
For aesthetic reasons, I think this should be the last field in the struct?

Reply:
Okay.